src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java

```diff
@@ -0,0 +1,249 @@
+package api.requests;
+
+import static api.support.builders.RequestBuilder.OPEN_NOT_YET_FILLED;
+import static api.support.matchers.PatronNoticeMatcher.hasEmailNoticeProperties;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+import static org.joda.time.DateTimeZone.UTC;
+
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.awaitility.Awaitility;
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.hamcrest.Matcher;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.CheckInByBarcodeRequestBuilder;
+import api.support.builders.HoldingBuilder;
+import api.support.builders.ItemBuilder;
+import api.support.builders.NoticeConfigurationBuilder;
+import api.support.builders.NoticePolicyBuilder;
+import api.support.builders.RequestBuilder;
+import api.support.fixtures.ItemExamples;
+import api.support.fixtures.TemplateContextMatchers;
+import api.support.http.InventoryItemResource;
+import io.vertx.core.json.JsonObject;
+
+public class RequestScheduledNoticesProcessingTests extends APITests {
```

Comment:
Is there a test for when a hold expiration notice is scheduled for `upon/at`?

Reply:
Added a test when a hold expiration notice is scheduled for upon/at.